### ISSUE TITLE
Versioning via "Clone to New"

### ISSUE THREAD

**Issue by Gazook89** [naturalcrit/homebrewery#3186]
### Your idea:

This is sort of a half-baked idea, apologies in advance.

Right now one of the 'quickest' methods of making versions of brews is to just "Clone to New" your brew and save that with a new name like ".... v2.3".  

What if when doing "Clone to New" there was some UI for noting it as a "version" of the current brew?  In practical terms this would then give some input option for a version number (or auto-increment it somehow), but more importantly it would associate the brews together.  Then, in the User's "list page" those brews would be visually grouped together.  

The visual grouping could just be the same as any brew that has the same name has now (meaning they are just ordered together).  Or the brew 'cards' could appear as a stack, with a button to rotate through versions.  Or only brew item shows in the list, but when you open it you can navigate through versions via the Properties pane (or the meta info tooltip from the top toolbar in the Share page).  

Each brew version would in fact be it's own brew...it's no real version control system.  The author could still make changes to older versions.  Likely this is where things break down...like you'd probably have to prevent "cloning to new version" more than once on each brew.  

The strength or amount of 'association' between the versions could be very basic as described above, or you could more closely tie them together--- for example, if you change the title of the most recent version, it could cascade to the rest of them.

Just an idea.

Possibly ties into #1101, not as a solution but just as a thought semi-related to versioning.

**Comment by 5e-Cleric** [comment]
related to this, we could have the "Clone to new" button in the edit page as well, i have found myself in the need for it.

**Comment by lucastucious** [comment]
I don't think it should be a "new brew". We now have a page with version number and last input. Maybe add a feature to make "named versions", like v1, v2, v3 and de could access them from the version/info of the share brew.

**Comment by 5e-Cleric** [comment]
That is an interesting points, instead of having them as separate brews, be all accessible through one brew.

But, we kind of will have this sorted out the other way, with separate brews and then Folders, to group brews.

**Comment by dbolack-ab** [comment]
This is much like some of the talks I've had on the notion of document versioning over the past couple of years.
